### PR TITLE
Show current keyboard layout using libxkb

### PR DESCRIPTION
**Pull request by Ma27**
I currently use two keyboard layouts: a german one for docs and an american one for coding and I switch them regularly. Therefore I think that adding an indicator for the current XKB layout might be helpful.

### CODE REVIEW

**Comment by NBonaparte** [comment]
Would it be possible to use xcb-xkb instead of xlib? That way, we won't have to add additional dependencies.

**Comment by Ma27** [comment]
good point, will have a look at it :-)

File: examples/example.config [hunk 1]
```diff
@@ -205,4 +205,8 @@ topbar:{
 		type: "periodic";
 		internal-option1: "spotify";
 	}
+  keyboard: {
```
Comment (Ma27) [inline examples/example.config]: this looks like a tabs vs spaces problem :-)

**Comment by Ma27** [comment]
@NBonaparte just managed to find some time to rewrite the indicator using `xcb-xkb`, would be cool if you could have another look at it %)

File: Makefile [hunk 2]
```diff
@@ -4,7 +4,7 @@ CPPFLAGS += -DVERSION=\"$(VERSION)\" -D_POSIX_C_SOURCE=199309L -DYA_INTERNAL -DY
 			-DYA_BSPWM
 CFLAGS += -std=c99 -Iinclude -pedantic -Wall -flto -O2 `pkg-config --cflags pango pangocairo libconfig gdk-pixbuf-2.0 alsa`
 
-DEPS += pango pangocairo libconfig gdk-pixbuf-2.0 alsa
+DEPS += pango pangocairo libconfig gdk-pixbuf-2.0 alsa xkbcommon xkbcommon-x11 xcb-xkb
```
Comment (NBonaparte) [inline Makefile]: These can probably be included with `LDLIBS` (`-lxkbcommon -lxkbcommon-x11 -lxcb-xkb`).

File: src/intern_blks/ya_intern.c [hunk 3]
```diff
@@ -136,6 +138,56 @@ __attribute__ ((gnu_inline)) inline void ya_setup_prefix_suffix(ya_block_t *blk,
...
+
+	ctx = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
+	if (!ctx) ya_block_error(blk, "Cannot create new XCB context!");
+	device_id = xkb_x11_get_core_keyboard_device_id(ya.c);
+	if (device_id == -1) ya_block_error(blk, "Empty reply, cannot get device id!");
+
+	while (1) {
+		keymap = xkb_x11_keymap_new_from_device(ctx, ya.c, device_id, XKB_KEYMAP_COMPILE_NO_FLAGS);
+		if (!keymap) ya_block_error(blk, "Cannot find keymap!");
+		state = xkb_state_new(keymap);
```
Comment (NBonaparte) [inline src/intern_blks/ya_intern.c]: Shouldn't `xkb_state_unref(state)` be called after printing?

**Comment by Ma27** [comment]
@NBonaparte done :)

**Comment by NBonaparte** [comment]
Just one more thing: could you add `libxkbcommon-devel libxkb-common-x11-devel` to the Fedora dependencies and `libxkbcommon-dev libxkbcommon-x11-dev libxcb-xkb-dev` to the Ubuntu dependencies in the README?

**Comment by Ma27** [comment]
done, thanks a lot for your feedback :)

> could you add `libxkbcommon-devel libxkbcommon-x11-devel` to the Fedora dependencies

I'm no Fedora expert, but how do Fedora folks obtain `libxcb-xkb-devel` (didn't find such a package here: https://apps.fedoraproject.org/packages/s/xcb)

**Comment by NBonaparte** [comment]
It should be included in `libxcb-devel`: https://apps.fedoraproject.org/packages/libxcb-devel/

**Comment by Ma27** [comment]
ok good to know. Is there anything else tbd before merging?:)

**Pull request merged**